worker/undertaker/manifold.go

```diff
@@ -18,14 +18,14 @@ import (
 // ManifoldConfig holds the names of the resources used by, and the
 // additional dependencies of, an undertaker worker.
 type ManifoldConfig struct {
-	APICallerName      string
-	CloudDestroyerName string
+	APICallerName string
 
 	Logger                       Logger
 	Clock                        clock.Clock
 	NewFacade                    func(base.APICaller) (Facade, error)
 	NewWorker                    func(Config) (worker.Worker, error)
 	NewCredentialValidatorFacade func(base.APICaller) (common.CredentialAPI, error)
+	NewEnvironFunc               environs.NewEnvironFunc
```

Comment:
We use CloudDestroyer because this worker needs to work for both caas and iaas and the CloudDestroyer represents common methods for those. caas doesn't use an environ.